WordPress/Classes/Utility/DiskCache.swift

```diff
@@ -0,0 +1,236 @@
+import Foundation
+import CryptoKit
+
+/// An LRU disk cache that stores data in files on disk.
+///
+/// ``DiskCache`` uses an LRU cleanup policy where the least recently used items
+/// are removed first). The elements stored in the cache are automatically
+/// discarded if either *cost* or *count* limit is exceeded.
+///
+/// ``DiskCache`` is thread-safe; both reads and writes can be executed in parallel.
+///
+/// - important: It's not recommended to have more than one instance of ``DiskCache``
+/// managing the same path.
+final class DiskCache {
+    static let shared = DiskCache(
+        url: URL.getCachesURL().appendingPathComponent("WPCache", isDirectory: true)
+    )
+
+    private let rootURL: URL
+    private let sizeLimit: Int
+    private let sweepInterval: TimeInterval = 86400 // Around 1 day
+
+    private let queue = DispatchQueue(label: "org.wordpress.diskCache")
+
+    /// Creates a cache instance with a given root URL.
+    ///
+    /// - parameters:
+    ///   - url: The directory URL where cached files will be stored. The
+    ///     directory will be automatically created by ``DiskCache``.
+    ///   - sizeLimit: The size limit in bytes. By default, 100 MB.
+    init(url: URL, sizeLimit: Int = 1024 * 1024 * 100) {
```

Comment:
Being flexible is good. Accepting all URL leaves lots of room for accidental errors though. For example, what if the url is a file, not a directory? What if the url is the `Documents`, or `Library` directory, or any directory that's not a _dedicated_ cache directory? Their files may get deleted by mistake because all the content are treated as "cache files" by the `sweep` function.

Reply:
I think it's a bit of a stretch to say that it leaves a ton of room for errors. There is a pre-configured shared instance that everyone should use. You will not be creating caches left and right in the app, and if you do, you'd approach it carefully. It's not unprecedented to have an API that accepts a URL in the initializer. See Core Data, `URLCache`.

I agree with the argument that it should _probably_ enforce that the files are stored in the caches directory, which is cleared automatically by the system when needed and is guaranteed not to be backed up in iCloud. But I think it can be up to the developer not to make a mistake like that.

The URL is still more flexible in case you want to put the cache in a subfolder in .caches. I don't feel strongly about it and I can bring back the original convenience initializer with a `name` parameter (renamed to something?)`.